src/app_service/service/token/async_tasks.nim

```diff
@@ -9,25 +9,29 @@ import ../../../backend/backend as backend
 
 type
   GetTokenDetailsTaskArg = ref object of QObjectTaskArg
-    chainId: int
+    chainIds: seq[int]
     address: string
 
 
 const getTokenDetailsTask*: Task = proc(argEncoded: string) {.gcsafe, nimcall.} =
   let arg = decode[GetTokenDetailsTaskArg](argEncoded)
-  try:
-    let response = backend.discoverToken(arg.chainId, arg.address).result
+  for chainId in arg.chainIds:
+    try:
+      let response = backend.discoverToken(chainId, arg.address).result
+      
+      let output = %* {
+        "address": arg.address,
+        "name": response{"name"}.getStr,
+        "symbol": response{"symbol"}.getStr,
+        "decimals": response{"decimals"}.getInt
+      }
+      arg.finish(output)
+      return
```

Comment:
Here the intention is to return as soon as there is on chain id that find the token. The logic is as follow:
for chain in allChain:
  if chain has token:
    return it
  else:
   continue

The try/except is doing this

Reply:
Aha... so we expect that `backend.discoverToken` raise an exception in case there are no tokens for passed `chainId` and `address`... in that case I would rather expect `error` param of response to be set instead.